File: Microsoft.VisualStudio.ServiceModel.DomainServices.Tools/Test/Baselines/EF_Northwind_Scenarios.vb

Option Compare Binary
Option Infer On
Option Strict On
Option Explicit On

Imports DataTests.Scenarios.EF.Northwind
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations
Imports System.Data
Imports System.Linq
Imports System.ServiceModel.DomainServices.EntityFramework
Imports System.ServiceModel.DomainServices.Hosting
Imports System.ServiceModel.DomainServices.Server

Namespace BizLogic.Test
    
    'Implements application logic using the NorthwindEntities_Scenarios context.
    ' TODO: Add your application logic to these methods or in additional methods.
    ' TODO: Wire up authentication (Windows/ASP.NET Forms) and uncomment the following to disable anonymous access
    ' Also consider adding roles to restrict access as appropriate.
    '<RequiresAuthentication> _
    <EnableClientAccess()>  _
    Public Class EF_Northwind_Scenarios
        Inherits LinqToEntitiesDomainService(Of NorthwindEntities_Scenarios)
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        'To support paging you will need to add ordering to the 'CustomerSet' query.
        Public Function GetCustomerSet() As IQueryable(Of Customer)
            Return Me.ObjectContext.CustomerSet
        End Function
        
        Public Sub InsertCustomer(ByVal customer As Customer)
            If ((customer.EntityState = EntityState.Detached)  _
                        = false) Then
                Me.ObjectContext.ObjectStateManager.ChangeObjectState(customer, EntityState.Added)
            Else
                Me.ObjectContext.CustomerSet.AddObject(customer)
            End If
        End Sub
        
        Public Sub UpdateCustomer(ByVal currentCustomer As Customer)
            Me.ObjectContext.CustomerSet.AttachAsModified(currentCustomer, Me.ChangeSet.GetOriginal(currentCustomer))
        End Sub
        
        Public Sub DeleteCustomer(ByVal customer As Customer)
            If ((customer.EntityState = EntityState.Detached)  _
                        = false) Then
                Me.ObjectContext.ObjectStateManager.ChangeObjectState(customer, EntityState.Deleted)
            Else
                Me.ObjectContext.CustomerSet.Attach(customer)
                Me.ObjectContext.CustomerSet.DeleteObject(customer)
            End If
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        'To support paging you will need to add ordering to the 'EmployeeSet' query.
        Public Function GetEmployeeSet() As IQueryable(Of Employee)
            Return Me.ObjectContext.EmployeeSet
        End Function
        
        Public Sub InsertEmployee(ByVal employee As Employee)
            If ((employee.EntityState = EntityState.Detached)  _
                        = false) Then
                Me.ObjectContext.ObjectStateManager.ChangeObjectState(employee, EntityState.Added)
            Else
                Me.ObjectContext.EmployeeSet.AddObject(employee)
            End If
        End Sub
        
        Public Sub UpdateEmployee(ByVal currentEmployee As Employee)
            Me.ObjectContext.EmployeeSet.AttachAsModified(currentEmployee, Me.ChangeSet.GetOriginal(currentEmployee))
        End Sub
        
        Public Sub DeleteEmployee(ByVal employee As Employee)
            If ((employee.EntityState = EntityState.Detached)  _
                        = false) Then
                Me.ObjectContext.ObjectStateManager.ChangeObjectState(employee, EntityState.Deleted)
            Else
                Me.ObjectContext.EmployeeSet.Attach(employee)
                Me.ObjectContext.EmployeeSet.DeleteObject(employee)
            End If
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        'To support paging you will need to add ordering to the 'EmployeeWithCTs' query.
        Public Function GetEmployeeWithCTs() As IQueryable(Of EmployeeWithCT)
            Return Me.ObjectContext.EmployeeWithCTs
        End Function
        
        Public Sub InsertEmployeeWithCT(ByVal employeeWithCT As EmployeeWithCT)
            If ((employeeWithCT.EntityState = EntityState.Detached)  _
                        = false) Then
                Me.ObjectContext.ObjectStateManager.ChangeObjectState(employeeWithCT, EntityState.Added)
            Else
                Me.ObjectContext.EmployeeWithCTs.AddObject(employeeWithCT)
            End If
        End Sub
        
        Public Sub UpdateEmployeeWithCT(ByVal currentEmployeeWithCT As EmployeeWithCT)
            Me.ObjectContext.EmployeeWithCTs.AttachAsModified(currentEmployeeWithCT, Me.ChangeSet.GetOriginal(currentEmployeeWithCT))
        End Sub
        
        Public Sub DeleteEmployeeWithCT(ByVal employeeWithCT As EmployeeWithCT)
            If ((employeeWithCT.EntityState = EntityState.Detached)  _
                        = false) Then
                Me.ObjectContext.ObjectStateManager.ChangeObjectState(employeeWithCT, EntityState.Deleted)
            Else
                Me.ObjectContext.EmployeeWithCTs.Attach(employeeWithCT)
                Me.ObjectContext.EmployeeWithCTs.DeleteObject(employeeWithCT)
            End If
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        'To support paging you will need to add ordering to the 'EntitiesWithNullFacetValuesForTimestampComparison' query.
        Public Function GetEntitiesWithNullFacetValuesForTimestampComparison() As IQueryable(Of EntityWithNullFacetValuesForTimestampComparison)
            Return Me.ObjectContext.EntitiesWithNullFacetValuesForTimestampComparison
        End Function
        
        Public Sub InsertEntityWithNullFacetValuesForTimestampComparison(ByVal entityWithNullFacetValuesForTimestampComparison As EntityWithNullFacetValuesForTimestampComparison)
            If ((entityWithNullFacetValuesForTimestampComparison.EntityState = EntityState.Detached)  _
                        = false) Then
                Me.ObjectContext.ObjectStateManager.ChangeObjectState(entityWithNullFacetValuesForTimestampComparison, EntityState.Added)
            Else
                Me.ObjectContext.EntitiesWithNullFacetValuesForTimestampComparison.AddObject(entityWithNullFacetValuesForTimestampComparison)
            End If
        End Sub
        
        Public Sub UpdateEntityWithNullFacetValuesForTimestampComparison(ByVal currentEntityWithNullFacetValuesForTimestampComparison As EntityWithNullFacetValuesForTimestampComparison)
            Me.ObjectContext.EntitiesWithNullFacetValuesForTimestampComparison.AttachAsModified(currentEntityWithNullFacetValuesForTimestampComparison)
        End Sub
        
        Public Sub DeleteEntityWithNullFacetValuesForTimestampComparison(ByVal entityWithNullFacetValuesForTimestampComparison As EntityWithNullFacetValuesForTimestampComparison)
            If ((entityWithNullFacetValuesForTimestampComparison.EntityState = EntityState.Detached)  _
                        = false) Then
                Me.ObjectContext.ObjectStateManager.ChangeObjectState(entityWithNullFacetValuesForTimestampComparison, EntityState.Deleted)
            Else
                Me.ObjectContext.EntitiesWithNullFacetValuesForTimestampComparison.Attach(entityWithNullFacetValuesForTimestampComparison)
                Me.ObjectContext.EntitiesWithNullFacetValuesForTimestampComparison.DeleteObject(entityWithNullFacetValuesForTimestampComparison)
            End If
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        'To support paging you will need to add ordering to the 'RequiredAttributeTestEntities' query.
        Public Function GetRequiredAttributeTestEntities() As IQueryable(Of RequiredAttributeTestEntity)
            Return Me.ObjectContext.RequiredAttributeTestEntities
        End Function
        
        Public Sub InsertRequiredAttributeTestEntity(ByVal requiredAttributeTestEntity As RequiredAttributeTestEntity)
            If ((requiredAttributeTestEntity.EntityState = EntityState.Detached)  _
                        = false) Then
                Me.ObjectContext.ObjectStateManager.ChangeObjectState(requiredAttributeTestEntity, EntityState.Added)
            Else
                Me.ObjectContext.RequiredAttributeTestEntities.AddObject(requiredAttributeTestEntity)
            End If
        End Sub
        
        Public Sub UpdateRequiredAttributeTestEntity(ByVal currentRequiredAttributeTestEntity As RequiredAttributeTestEntity)
            Me.ObjectContext.RequiredAttributeTestEntities.AttachAsModified(currentRequiredAttributeTestEntity, Me.ChangeSet.GetOriginal(currentRequiredAttributeTestEntity))
        End Sub
        
        Public Sub DeleteRequiredAttributeTestEntity(ByVal requiredAttributeTestEntity As RequiredAttributeTestEntity)
            If ((requiredAttributeTestEntity.EntityState = EntityState.Detached)  _
                        = false) Then
                Me.ObjectContext.ObjectStateManager.ChangeObjectState(requiredAttributeTestEntity, EntityState.Deleted)
            Else
                Me.ObjectContext.RequiredAttributeTestEntities.Attach(requiredAttributeTestEntity)
                Me.ObjectContext.RequiredAttributeTestEntities.DeleteObject(requiredAttributeTestEntity)
            End If
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        'To support paging you will need to add ordering to the 'TimestampEntities' query.
        Public Function GetTimestampEntities() As IQueryable(Of TimestampEntity)
            Return Me.ObjectContext.TimestampEntities
        End Function
        
        Public Sub InsertTimestampEntity(ByVal timestampEntity As TimestampEntity)
            If ((timestampEntity.EntityState = EntityState.Detached)  _
                        = false) Then
                Me.ObjectContext.ObjectStateManager.ChangeObjectState(timestampEntity, EntityState.Added)
            Else
                Me.ObjectContext.TimestampEntities.AddObject(timestampEntity)
            End If
        End Sub
        
        Public Sub UpdateTimestampEntity(ByVal currentTimestampEntity As TimestampEntity)
            Me.ObjectContext.TimestampEntities.AttachAsModified(currentTimestampEntity)
        End Sub
        
        Public Sub DeleteTimestampEntity(ByVal timestampEntity As TimestampEntity)
            If ((timestampEntity.EntityState = EntityState.Detached)  _
                        = false) Then
                Me.ObjectContext.ObjectStateManager.ChangeObjectState(timestampEntity, EntityState.Deleted)
            Else
                Me.ObjectContext.TimestampEntities.Attach(timestampEntity)
                Me.ObjectContext.TimestampEntities.DeleteObject(timestampEntity)
            End If
        End Sub
    End Class
End Namespace

File: OpenRiaServices.VisualStudio.DomainServices.Tools/Test/Baselines/EF_Northwind_Scenarios.cs

namespace BizLogic.Test
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.ComponentModel.DataAnnotations;
    using System.Data;
    using System.Linq;
    using OpenRiaServices.DomainServices.EntityFramework;
    using OpenRiaServices.DomainServices.Hosting;
    using OpenRiaServices.DomainServices.Server;
    using DataTests.Scenarios.EF.Northwind;
    
    
    // Implements application logic using the NorthwindEntities_Scenarios context.
    // TODO: Add your application logic to these methods or in additional methods.
    // TODO: Wire up authentication (Windows/ASP.NET Forms) and uncomment the following to disable anonymous access
    // Also consider adding roles to restrict access as appropriate.
    // [RequiresAuthentication]
    [EnableClientAccess()]
    public class EF_Northwind_Scenarios : LinqToEntitiesDomainService<NorthwindEntities_Scenarios>
    {
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        // To support paging you will need to add ordering to the 'CustomerSet' query.
        public IQueryable<Customer> GetCustomerSet()
        {
            return this.ObjectContext.CustomerSet;
        }
        
        public void InsertCustomer(Customer customer)
        {
            if ((customer.EntityState != EntityState.Detached))
            {
                this.ObjectContext.ObjectStateManager.ChangeObjectState(customer, EntityState.Added);
            }
            else
            {
                this.ObjectContext.CustomerSet.AddObject(customer);
            }
        }
        
        public void UpdateCustomer(Customer currentCustomer)
        {
            this.ObjectContext.CustomerSet.AttachAsModified(currentCustomer, this.ChangeSet.GetOriginal(currentCustomer));
        }
        
        public void DeleteCustomer(Customer customer)
        {
            if ((customer.EntityState != EntityState.Detached))
            {
                this.ObjectContext.ObjectStateManager.ChangeObjectState(customer, EntityState.Deleted);
            }
            else
            {
                this.ObjectContext.CustomerSet.Attach(customer);
                this.ObjectContext.CustomerSet.DeleteObject(customer);
            }
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        // To support paging you will need to add ordering to the 'EmployeeSet' query.
        public IQueryable<Employee> GetEmployeeSet()
        {
            return this.ObjectContext.EmployeeSet;
        }
        
        public void InsertEmployee(Employee employee)
        {
            if ((employee.EntityState != EntityState.Detached))
            {
                this.ObjectContext.ObjectStateManager.ChangeObjectState(employee, EntityState.Added);
            }
            else
            {
                this.ObjectContext.EmployeeSet.AddObject(employee);
            }
        }
        
        public void UpdateEmployee(Employee currentEmployee)
        {
            this.ObjectContext.EmployeeSet.AttachAsModified(currentEmployee, this.ChangeSet.GetOriginal(currentEmployee));
        }
        
        public void DeleteEmployee(Employee employee)
        {
            if ((employee.EntityState != EntityState.Detached))
            {
                this.ObjectContext.ObjectStateManager.ChangeObjectState(employee, EntityState.Deleted);
            }
            else
            {
                this.ObjectContext.EmployeeSet.Attach(employee);
                this.ObjectContext.EmployeeSet.DeleteObject(employee);
            }
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        // To support paging you will need to add ordering to the 'EmployeeWithCTs' query.
        public IQueryable<EmployeeWithCT> GetEmployeeWithCTs()
        {
            return this.ObjectContext.EmployeeWithCTs;
        }
        
        public void InsertEmployeeWithCT(EmployeeWithCT employeeWithCT)
        {
            if ((employeeWithCT.EntityState != EntityState.Detached))
            {
                this.ObjectContext.ObjectStateManager.ChangeObjectState(employeeWithCT, EntityState.Added);
            }
            else
            {
                this.ObjectContext.EmployeeWithCTs.AddObject(employeeWithCT);
            }
        }
        
        public void UpdateEmployeeWithCT(EmployeeWithCT currentEmployeeWithCT)
        {
            this.ObjectContext.EmployeeWithCTs.AttachAsModified(currentEmployeeWithCT, this.ChangeSet.GetOriginal(currentEmployeeWithCT));
        }
        
        public void DeleteEmployeeWithCT(EmployeeWithCT employeeWithCT)
        {
            if ((employeeWithCT.EntityState != EntityState.Detached))
            {
                this.ObjectContext.ObjectStateManager.ChangeObjectState(employeeWithCT, EntityState.Deleted);
            }
            else
            {
                this.ObjectContext.EmployeeWithCTs.Attach(employeeWithCT);
                this.ObjectContext.EmployeeWithCTs.DeleteObject(employeeWithCT);
            }
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        // To support paging you will need to add ordering to the 'EntitiesWithNullFacetValuesForTimestampComparison' query.
        public IQueryable<EntityWithNullFacetValuesForTimestampComparison> GetEntitiesWithNullFacetValuesForTimestampComparison()
        {
            return this.ObjectContext.EntitiesWithNullFacetValuesForTimestampComparison;
        }
        
        public void InsertEntityWithNullFacetValuesForTimestampComparison(EntityWithNullFacetValuesForTimestampComparison entityWithNullFacetValuesForTimestampComparison)
        {
            if ((entityWithNullFacetValuesForTimestampComparison.EntityState != EntityState.Detached))
            {
                this.ObjectContext.ObjectStateManager.ChangeObjectState(entityWithNullFacetValuesForTimestampComparison, EntityState.Added);
            }
            else
            {
                this.ObjectContext.EntitiesWithNullFacetValuesForTimestampComparison.AddObject(entityWithNullFacetValuesForTimestampComparison);
            }
        }
        
        public void UpdateEntityWithNullFacetValuesForTimestampComparison(EntityWithNullFacetValuesForTimestampComparison currentEntityWithNullFacetValuesForTimestampComparison)
        {
            this.ObjectContext.EntitiesWithNullFacetValuesForTimestampComparison.AttachAsModified(currentEntityWithNullFacetValuesForTimestampComparison);
        }
        
        public void DeleteEntityWithNullFacetValuesForTimestampComparison(EntityWithNullFacetValuesForTimestampComparison entityWithNullFacetValuesForTimestampComparison)
        {
            if ((entityWithNullFacetValuesForTimestampComparison.EntityState != EntityState.Detached))
            {
                this.ObjectContext.ObjectStateManager.ChangeObjectState(entityWithNullFacetValuesForTimestampComparison, EntityState.Deleted);
            }
            else
            {
                this.ObjectContext.EntitiesWithNullFacetValuesForTimestampComparison.Attach(entityWithNullFacetValuesForTimestampComparison);
                this.ObjectContext.EntitiesWithNullFacetValuesForTimestampComparison.DeleteObject(entityWithNullFacetValuesForTimestampComparison);
            }
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        // To support paging you will need to add ordering to the 'RequiredAttributeTestEntities' query.
        public IQueryable<RequiredAttributeTestEntity> GetRequiredAttributeTestEntities()
        {
            return this.ObjectContext.RequiredAttributeTestEntities;
        }
        
        public void InsertRequiredAttributeTestEntity(RequiredAttributeTestEntity requiredAttributeTestEntity)
        {
            if ((requiredAttributeTestEntity.EntityState != EntityState.Detached))
            {
                this.ObjectContext.ObjectStateManager.ChangeObjectState(requiredAttributeTestEntity, EntityState.Added);
            }
            else
            {
                this.ObjectContext.RequiredAttributeTestEntities.AddObject(requiredAttributeTestEntity);
            }
        }
        
        public void UpdateRequiredAttributeTestEntity(RequiredAttributeTestEntity currentRequiredAttributeTestEntity)
        {
            this.ObjectContext.RequiredAttributeTestEntities.AttachAsModified(currentRequiredAttributeTestEntity, this.ChangeSet.GetOriginal(currentRequiredAttributeTestEntity));
        }
        
        public void DeleteRequiredAttributeTestEntity(RequiredAttributeTestEntity requiredAttributeTestEntity)
        {
            if ((requiredAttributeTestEntity.EntityState != EntityState.Detached))
            {
                this.ObjectContext.ObjectStateManager.ChangeObjectState(requiredAttributeTestEntity, EntityState.Deleted);
            }
            else
            {
                this.ObjectContext.RequiredAttributeTestEntities.Attach(requiredAttributeTestEntity);
                this.ObjectContext.RequiredAttributeTestEntities.DeleteObject(requiredAttributeTestEntity);
            }
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        // To support paging you will need to add ordering to the 'TimestampEntities' query.
        public IQueryable<TimestampEntity> GetTimestampEntities()
        {
            return this.ObjectContext.TimestampEntities;
        }
        
        public void InsertTimestampEntity(TimestampEntity timestampEntity)
        {
            if ((timestampEntity.EntityState != EntityState.Detached))
            {
                this.ObjectContext.ObjectStateManager.ChangeObjectState(timestampEntity, EntityState.Added);
            }
            else
            {
                this.ObjectContext.TimestampEntities.AddObject(timestampEntity);
            }
        }
        
        public void UpdateTimestampEntity(TimestampEntity currentTimestampEntity)
        {
            this.ObjectContext.TimestampEntities.AttachAsModified(currentTimestampEntity);
        }
        
        public void DeleteTimestampEntity(TimestampEntity timestampEntity)
        {
            if ((timestampEntity.EntityState != EntityState.Detached))
            {
                this.ObjectContext.ObjectStateManager.ChangeObjectState(timestampEntity, EntityState.Deleted);
            }
            else
            {
                this.ObjectContext.TimestampEntities.Attach(timestampEntity);
                this.ObjectContext.TimestampEntities.DeleteObject(timestampEntity);
            }
        }
    }
}
